src/julia_internal.h

```diff
@@ -678,6 +678,9 @@ JL_DLLEXPORT void jl_binding_deprecation_warning(jl_module_t *m, jl_binding_t *b
 extern jl_array_t *jl_module_init_order JL_GLOBALLY_ROOTED;
 extern htable_t jl_current_modules JL_GLOBALLY_ROOTED;
 extern JL_DLLEXPORT jl_module_t *jl_precompile_toplevel_module JL_GLOBALLY_ROOTED;
+extern jl_array_t *jl_global_roots_table JL_GLOBALLY_ROOTED;
+JL_DLLEXPORT int jl_is_globally_rooted(jl_value_t *val);
+JL_DLLEXPORT jl_value_t *jl_as_global_root(jl_value_t *val);
```

Comment:
Is this argument JL_MAYBE_UNROOTED, or why does it have a root created inside?

Reply:
Yes, there is at least one place in codegen that calls it with an unrooted argument. I'll add the annotation.